samples/Rules.Framework.InMemory.Sample/Rules/TestNumberRules.cs

```diff
@@ -27,41 +27,41 @@ public IEnumerable<RuleSpecification> GetRulesSpecifications()
         }
 
         private void Add(
-            RuleBuilderResult<ContentTypes, ConditionTypes> rule,
+            RuleBuilderResult<RulesetNames, ConditionNames> rule,
             RuleAddPriorityOption ruleAddPriorityOption) => rulesSpecifications.Add(
                 new RuleSpecification
                 {
                     RuleBuilderResult = rule,
                     RuleAddPriorityOption = ruleAddPriorityOption,
                 });
 
-        private RuleBuilderResult<ContentTypes, ConditionTypes> CreateDefaultRule() => Rule.New<ContentTypes, ConditionTypes>()
-            .WithName("Default rule for test number")
-            .WithContent(ContentTypes.TestNumber, ":| default nothing special about this number")
-            .WithDateBegin(new DateTime(2019, 01, 01))
+        private RuleBuilderResult<RulesetNames, ConditionNames> CreateDefaultRule() => Rule.Create<RulesetNames, ConditionNames>("Default rule for test number")
+            .OnRuleset(RulesetNames.TestNumber)
+            .SetContent(":| default nothing special about this number")
+            .Since(new DateTime(2019, 01, 01))
             .Build();
 
-        private RuleBuilderResult<ContentTypes, ConditionTypes> CreateRuleForCoolNumbers() => Rule.New<ContentTypes, ConditionTypes>()
-            .WithName("Rule for cool numbers")
-            .WithContent(ContentTypes.TestNumber, ":D this number is so COOL!")
-            .WithDateBegin(new DateTime(2019, 01, 01))
-            .WithCondition(c => c
-            .Or(o => o
-                .Value(ConditionTypes.RoyalNumber, Operators.Equal, 7)
-                .And(a => a
-                    .Value(ConditionTypes.IsPrimeNumber, Operators.Equal, 7)
-                    .Value(ConditionTypes.SumAll, Operators.StartsWith, "5"))))
+        private RuleBuilderResult<RulesetNames, ConditionNames> CreateRuleForCoolNumbers() => Rule.Create<RulesetNames, ConditionNames>("Rule for cool numbers")
+            .OnRuleset(RulesetNames.TestNumber)
+            .SetContent(":D this number is so COOL!")
+            .Since(new DateTime(2019, 01, 01))
```

Comment:
It could be nice to have an extension method that allows the dev to use a simple string to define the date or even use the args directly, like `.Since(2024,01,01)`. For the sake of the better reading.

Reply:
I agree, I will add to this PR 👍

Reply:
Please check `Scenario7Data` and `Scenario8Data` on the PR changed files to see examples of your suggestions.